Profile style

jdaniel0x50 opened the pull request:
Related to edit profile ticket
Note: This does NOT complete the ticket requirements. It is an initial iterative push to clean up the profile page styles so that Jenna (also working on the profile page) can incorporate these changes as she edits the sidebar.

**Changes in this PR include:**

- Adapted styles to make the profile page and components more responsive to different screen size
- Changed styles to bootstrap styles - and removed custom styles (I am still working on this - this is an initial step in this direction)
- Changed custom input fields to incorporate pre-built input components to handle validation (note - a future build will incorporate validation logic on the form so that invalid inputs will not permit the user to submit the form)

**This PR does not include**:

- Form validation (forbid incomplete or invalid data to submit) (it does, however, include individual input field validation)
- Form fields will default to show current data ... will be an easy fix, but need to incorporate
- Submits will only update fields that change
- Provide a confirmation to the user when data updates successfully
- Ability to see multiple children applications and status (on camp profile)

EDITS PUSHED ON FRIDAY, 8/3/2018

- Added form validation for name, email, and password fields
- Added validation to validate input fields and their confirm fields match
- Disabled submit if form fields are invalid or if confirm fields do not match
- Modified error messages
- NOTE: The form does not currently submit to the back end -- will need to modify the back-end routes and adjust this axios.post in a future PR

GLOBAL EDITS

- added a global cursor: pointer for all buttons and a tags in App.vue
- added a String.trim() function to the input component such that all inputs will not submit with leading or trailing spaces

jdaniel0x50 committed:
modified profile edit page to make responsive, incorporate pre-built input components --- also added new prop to the input components to be able to accept add on text and colors to include an input field styled add on with bootstrap

jdaniel0x50 committed:
Merge branch 'master' of https://github.com/the-difference-engine/we-r-next-generation-web into profile-style

jdaniel0x50 committed:
modified profile edit input fields with an edit button - users must click an edit checkbox to edit fields - this checkbox will be used to determine which fields will submit in an update

jdaniel0x50 committed:
modified profile page to include validations for name, email, password, including validation of confirm fields; disabled submit if form is invalid; added logic to submit original profile data if the user did not edit a name, email, or password field -- also added global pointer styling to buttons -- and added a trim function for all input fields in the input component

jdaniel0x50 committed:
small change to the input-field-only component to expand the input to full width after a small change i made in a previous commit broke this width

mkarazin commented on src/components/forms/inputFieldOnly.vue:
```diff
@@ -309,11 +334,17 @@ export default {
+			}
 			this.isPristine = false;
-			this.runValidators();
+			if (this.readonly == false) {
```
I'd recommend use of the identity check `===` over the use of equality `==`, for this and all other conditional checks.

jdaniel0x50 replied:
Sounds great. Made the changes and pushed to this PR

jdaniel0x50 committed:
modified conditionals to include an identity check '==='

jdaniel0x50 committed:
added a new children component to view all children that have an application on the parent profile page -- the subcomponent will group and sort submitted applications by child name

jdaniel0x50 commented:
UPDATE on Wednesday, August 8:

Added changes to include a list of all children that have applications on the parent profile. The children.vue sub component uses a new backend API route to group and sort all applications related to the logged in user profile by child name

The list of children, includes the child age and all associated applications for the child - including the application date and applicaiton status

There is also a link on the application date to view the submitted application

![image](https://user-images.githubusercontent.com/32044884/43874601-47223318-9b52-11e8-8368-21ea46ff81a7.png)

jdaniel0x50 committed:
modified the application submitted page to include application data in addition to the waiver

jdaniel0x50 committed:
small change to children.vue to filter out volunteer applications and only list camper applications

mkirlin commented on src/components/applications/children.vue:
```diff
@@ -0,0 +1,78 @@
+        localforage.getItem("X_TOKEN")
+        .then(session => {
+            this.sessionId = session;
+            axios.get("/api/v1/profile/" + session + "/applications", {
```
Hey, I altered the back-end pretty significantly. I implemented your changes from your last PR, but I tweaked some pieces of it. You can replace the url here with this: `"/api/v1/profile/applications"` and it should return the same thing.

jdaniel0x50 committed:
added address information to the profile edit page -- modified the route to get applications, per suggestion from Michael on a previous PR review -- divided profile page into multiple components to simplify code --> profile + view profile component + edit profile component

jdaniel0x50 commented:
I submitted a new revision to this PR this evening, including the following changes:

- Addressed the changes requested from Michael (re: the backend API route)
- Added address view and edit fields to the profile page (re: conversation on Slack) / note this does not change the profile address information on registration - this will be submitted in a separate PR
- Added email validation to the input component
- Divided the large profile component into three components: profile, profile view, profile edit

zachgarwood reviewed (approved):
LGTM 🦄
